planner/core/common_plans.go

```diff
@@ -732,19 +732,53 @@ func (e *Explain) RenderResult() error {
 }
 
 // explainPlanInRowFormat generates explain information for root-tasks.
-func (e *Explain) explainPlanInRowFormat(p Plan, taskType, indent string, isLastChild bool) (err error) {
-	e.prepareOperatorInfo(p, taskType, indent, isLastChild)
+func (e *Explain) explainPlanInRowFormat(p Plan, taskType, driverSide, indent string, isLastChild bool) (err error) {
+	e.prepareOperatorInfo(p, taskType, driverSide, indent, isLastChild)
 	e.explainedPlans[p.ID()] = true
 
 	// For every child we create a new sub-tree rooted by it.
 	childIndent := texttree.Indent4Child(indent, isLastChild)
 
 	if physPlan, ok := p.(PhysicalPlan); ok {
+		// indicate driven side and driving side of 'join'.
+		// See issue https://github.com/pingcap/tidb/issues/14602.
+		driverSideInfo := make([]string, len(physPlan.Children()))
+		buildSide := -1
+
+		switch physJoinPlan := physPlan.(type) {
+		case *PhysicalHashJoin:
+			if physJoinPlan.UseOuterToBuild {
+				buildSide = physJoinPlan.InnerChildIdx ^ 1
+			} else {
+				buildSide = physJoinPlan.InnerChildIdx
+			}
+		case *PhysicalMergeJoin:
+			if physJoinPlan.JoinType == RightOuterJoin {
+				buildSide = 0
+			} else {
+				buildSide = 1
+			}
+		case *PhysicalIndexJoin:
+			buildSide = physJoinPlan.InnerChildIdx ^ 1
+		case *PhysicalIndexMergeJoin:
+			buildSide = physJoinPlan.InnerChildIdx ^ 1
+		case *PhysicalIndexHashJoin:
+			buildSide = physJoinPlan.InnerChildIdx ^ 1
+		}
+
+		if buildSide != -1 {
+			if len(driverSideInfo) < 2 {
```

Comment:
IMHO, we don't need this check. We can make sure that `PhysicalJoin` always has two children.

Reply:
Thanks, I see.